### PR TITLE
Add AbsolutePath.asURL

### PR DESCRIPTION
**Pull request by krzyzanowskim**
We all use it as an extension anyway
see https://github.com/apple/sourcekit-lsp/pull/27/files#diff-c56368b1b05968c4effa47f617ce7434R19

### CODE REVIEW

**Comment by neonichu** [comment]
Don't think we should have this. `AbsolutePath` is an alternative API to `URL`, so most people won't need such an extension and it brings in Foundation. Not even `String` has `asURL`.

**Comment by krzyzanowskim** [comment]
AbsolutePath definitely is related to path expressed as a string, such path can be expressed as an URL especially fileURL. From my experience, whenever I use AbsolutePath, I have this kind of extension to deal with the outside world. Since this in lo longer an internal API, such outlet shouldn't do any harm

the Foundation shouldn't be the problem, it's not a new dependency in SwiftPM.

**Comment by aciidb0mb3r** [comment]
I think it is fine to have this extension in `AbsolutePath` since Foundation is a core library.

File: Sources/Basic/Path.swift [hunk 1]
```diff
@@ -205,6 +207,11 @@ public struct AbsolutePath: Hashable {
         return _impl.string
     }
 
+    /// File URL created from the normalized string representation of the path.
+    public var asURL: Foundation.URL {
```
Comment (aciidb0mb3r) [inline Sources/Basic/Path.swift]: Let's make this as an extension so we can avoid importing Foundation in this file. We can put the extension in `misc.swift`.

Comment (krzyzanowskim) [reply Sources/Basic/Path.swift]: done

File: Sources/Basic/misc.swift [hunk 2]
```diff
@@ -140,3 +140,10 @@ extension CodableRange: Codable {
         self.init(Range(uncheckedBounds: (lowerBound, upperBound)))
     }
 }
+
+extension AbsolutePath {
+    /// File URL created from the normalized string representation of the path.
+    public var asURL: Foundation.URL {
+         return URL(fileURLWithPath: asString)
+    }
+}
```
Comment (aciidb0mb3r) [inline Sources/Basic/misc.swift]: nit: missing new line

**Comment by neonichu** [comment]
@swift-ci please smoke test

**Comment by neonichu** [comment]
@swift-ci please smoke test

**Pull request merged**